### PR TITLE
Dynamic pages

### PR DESCRIPTION
**Pull request by sabrinababakulova**
THIS PROJECT IS IN DEVELOPMENT MODE.

card/:id added

## Description
Добавить новую страницу “/card/:id” При открытии страницы “/card/:id” на странице должен отобразится заголовок карточки, её текст и возможность редактировать заголовок и текст. При двойном клике на карточку в режиме, когда карточка в режиме просмотра (не редактировании), пользователь должен перейти на страницу “/card/:id

## Screenshots (if appropriate):
image
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-                    <Route path="/signin" element={<SignInPage />} />
-                    <Route path="*" element={<NotFoundPage />} />
-                </Routes>
-            </Box>
-        </Router>
-    )
+  const dispatch = useDispatch()
+  useEffect(() => {
+    dispatch(fetchData())
+  }, [])
```
Comment (vshynkarova) [inline src/App.js]: Why you did this dispatch in this component?

File: .prettierrc.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-    "trailingComma": "es5",
-    "tabWidth": 4,
-    "semi": false,
-    "singleQuote": true,
-    "printWidth": 80,
-    "useTabs": false,
-    "endOfLine": "auto"
+  "trailingComma": "es5",
+  "tabWidth": 2,
+  "semi": false,
```
Comment (vshynkarova) [inline .prettierrc.json]: Update to `semi: true`

File: src/components/Card/index.jsx [hunk 3]
```diff
@@ -6,113 +6,117 @@ import CardHeader from './CardHeader'
...
+  const checkedCards = useSelector((state) => state.cards.checkedCards)
+  const dispatch = useDispatch()
+  const boxChecked = checkedCards.includes(data.id)
+  const [header, setHeader] = useState(data.header)
+  const [body, setBody] = useState(data.body)
+  const [revertHeader, setRevertHeader] = useState(header)
+  const [revertBody, setRevertBody] = useState(body)
+  const [isEditing, setIsEditing] = useState(editing)
+  const [editApproved, setEditApproved] = useState(true)
+  const handleDOubleClick = () => {
```
Comment (vshynkarova) [inline src/components/Card/index.jsx]: Typo: handleDoubleClick

File: src/components/Card/index.jsx [hunk 4]
```diff
@@ -6,113 +6,117 @@ import CardHeader from './CardHeader'
...
+  const validationOnSave = () => {
+    if (validator.isEmpty(header, { ignore_whitespace: true })) {
+      setIsEditing(true)
+      setEditApproved(false)
+    } else {
+      if (newCard) {
+        const createdCard = {
+          body: body,
+          header: header,
+          id: uuidv4(),
```
Comment (vshynkarova) [inline src/components/Card/index.jsx]: Same names can be omitted: 
`{
          body,
          header,
          id: uuidv4()
 }`

File: src/components/Card/index.jsx [hunk 5]
```diff
@@ -6,113 +6,117 @@ import CardHeader from './CardHeader'
...
-    editing: PropTypes.bool,
-    newCard: PropTypes.bool,
-    onClose: PropTypes.func,
+  data: PropTypes.object,
+  editing: PropTypes.bool,
+  newCard: PropTypes.bool,
+  onClose: PropTypes.func,
 }
 
 export default CardWithLoadingDelay
```
Comment (vshynkarova) [inline src/components/Card/index.jsx]: No need to fix, just for information: More sommon way to use export: `export default withLoadingDelay(Card)`.

File: src/components/CardList/index.jsx [hunk 6]
```diff
@@ -1,67 +1,30 @@
...
+        mt="4"
+        templateColumns={['1fr', '1fr', '1fr', 'repeat(2, 1fr)']}
+        gap={8}
+      >
+        {cards.numberOfCards === 0 ? (
+          <Badge colorScheme="red" mt="4" fontSize="1em">
+            You dont have any cards
+          </Badge>
+        ) : (
+          cards.cards
```
Comment (vshynkarova) [inline src/components/CardList/index.jsx]: `cards.cards`?))) Rename!!

File: src/components/DataFetching/index.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import axios from 'axios'
```
Comment (vshynkarova) [inline src/components/DataFetching/index.js]: Move this file in the `api` folder. 
Leave only ` axios.get`.
You don't need here `then` and `catch`.

File: src/components/Input/index.jsx [hunk 8]
```diff
@@ -2,34 +2,34 @@ import React, { useEffect, useState } from 'react'
 import PropTypes from 'prop-types'
```
Comment (vshynkarova) [inline src/components/Input/index.jsx]: Remove unused imports from this file

File: src/components/Input/index.jsx [hunk 9]
```diff
@@ -2,34 +2,34 @@ import React, { useEffect, useState } from 'react'
...
-    border-radius: 10px;
-    border: 1px solid teal;
-    padding: 10px;
-    margin: 10px;
-    width: 20rem;
+  border-radius: 10px;
+  border: 1px solid teal;
+  padding: 10px;
+  margin: 10px;
+  width: 20rem;
```
Comment (vshynkarova) [inline src/components/Input/index.jsx]: Move styled component in `styles` file and import from there

File: src/components/SignIn/index.jsx [hunk 10]
```diff
@@ -4,88 +4,88 @@ import validator from 'validator'
...
-        `
+  border-radius: 10px;
+  padding: 10px;
+  border: 2px solid teal;
+  ${(props) =>
+    props.disabled &&
+    `
         border-color: red;
         cursor: not-allowed;
   `}
```
Comment (vshynkarova) [inline src/components/SignIn/index.jsx]: Move styled components in styles file and import from there

File: src/components/SignIn/index.jsx [hunk 11]
```diff
@@ -4,88 +4,88 @@ import validator from 'validator'
...
-    const [password, setPassword] = useState('')
-    const [errorExist, setErrorExist] = useState(false)
-    const [errorMessage, setErrorMessage] = useState('')
-    const UserNameError = 'Please make sure that your user name is your email'
-    const PasswordError =
-        'Please make sure that the password is at least 8 characters long, and contains at least one number and one letter.'
+  const [userName, setUserName] = useState('')
+  const [password, setPassword] = useState('')
+  const [errorExist, setErrorExist] = useState(false)
+  const [errorMessage, setErrorMessage] = useState('')
```
Comment (vshynkarova) [inline src/components/SignIn/index.jsx]: Do you really need both of them?
`errorExist` is redundant.

File: src/store/AllCardsSlice/index.js [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+      state.status = 'loading'
+    },
+    [fetchData.fulfilled]: (state, action) => {
+      state.status = 'success'
+      state.cards = action.payload
+      state.numberOfCards = state.cards.length
+    },
+    [fetchData.rejected]: (state) => {
+      state.status = 'error'
+    },
```
Comment (vshynkarova) [inline src/store/AllCardsSlice/index.js]: Praise: you did what I mentioned. Great!

Now let's think: you have state statuses, how can you use this in your application?

**Review by vshynkarova — CHANGES_REQUESTED** [review]
This pull request looks much better than the first one, well done!

To fix: 
1. after login I want to be redirected to the main page.

2. I don't want to see placeholders in not edit mode.

Screenshot 2022-07-21 at 08 47 24


3. Set the check mark or update style. 

https://user-images.githubusercontent.com/96103612/180143000-ef895a3b-e54f-4066-b9de-115c9ade1057.mov

File: src/App.js (hunk 1 above)
Comment (sabrinababakulova) [reply src/App.js]: because otherwise the cards are not being loaded into store

File: src/components/CardList/index.jsx [hunk 13]
```diff
@@ -1,30 +1,37 @@
...
-import { useSelector } from 'react-redux'
+import React from 'react';
+import { Box, Grid, Badge, Spinner, Center } from '@chakra-ui/react';
+import Card from '../Card';
+import { useSelector } from 'react-redux';
 
 const CardList = () => {
-  const { cards } = useSelector((state) => state)
+  const { allCardsInfo } = useSelector((state) => state);
+  if (allCardsInfo.status === 'loading') {
```
Comment (vshynkarova) [inline src/components/CardList/index.jsx:12]: ❤️

**Review by vshynkarova — APPROVED** [review]

**Pull request merged**